x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_base.ts

```diff
@@ -60,19 +61,53 @@ export abstract class InferenceBase<TInferResponse> {
   protected abstract readonly inferenceTypeLabel: string;
   protected inputField: string;
   protected readonly modelInputField: string;
-  private inputText$ = new BehaviorSubject<string[]>([]);
+  protected inputText$ = new BehaviorSubject<string[]>([]);
   private inferenceResult$ = new BehaviorSubject<TInferResponse[] | null>(null);
   private inferenceError$ = new BehaviorSubject<MLHttpFetchError | null>(null);
   private runningState$ = new BehaviorSubject<RUNNING_STATE>(RUNNING_STATE.STOPPED);
+  protected inputTextValid$ = new BehaviorSubject<boolean>(false);
+  protected isValid$ = new BehaviorSubject<boolean>(false);
   protected readonly info: string[] = [];
 
+  private validators$: Array<Observable<boolean>> = [];
+  private validatorsSubscriptions$: Subscription = new Subscription();
+
+  private pipeline = new BehaviorSubject<estypes.IngestPipeline>({});
```

Comment:
I reckon we should follow the same naming convention for observables/subjects 
```suggestion
  private pipeline$ = new BehaviorSubject<estypes.IngestPipeline>({});
```